### ISSUE TITLE
Rule: Warn on bash-style array syntax for non-bash-family shells

### ISSUE THREAD

**Issue by mcandre** [koalaman/shellcheck#2575]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
echo $somearray[@]
echo $somearray[*]
```

These lines may be fine in bash, zsh, dash, posh, etc., but the lines are invalid in (POSIX) sh, ksh, and so on.

#### Here's what I wanted or expected to see:

Bash array syntax used, though a strict POSIX sh interpreter was detected.

**Comment by brother** [comment]
Isn't the current warning enough?

[SC3054](https://www.shellcheck.net/wiki/SC3054) (warning): In POSIX sh, array references are undefined.

**Comment by brother** [comment]
(you need to add the curly braces to get the error.)
[SC1087](https://www.shellcheck.net/wiki/SC1087) (error): Use braces when expanding arrays, e.g. ${array[idx]} (or ${var}[.. to quiet). )
